packages/braintree-integration/src/braintree-visa-checkout/braintree-visa-checkout-customer-strategy.ts

```diff
@@ -0,0 +1,253 @@
+                'Unable to proceed because "options.braintreevisacheckout" argument is not provided.',
+            );
+        }
+
+        await this.paymentIntegrationService.loadPaymentMethod(methodId);
+
+        const state = this.paymentIntegrationService.getState();
+
+        const checkout = state.getCheckout();
+
+        const { clientToken, config } = state.getPaymentMethodOrThrow(methodId);
+
+        const storeConfig = state.getStoreConfigOrThrow();
+
+        if (!checkout) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+        }
+
+        if (!storeConfig) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+        }
+
+        if (!clientToken) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this.braintreeSdk.initialize(clientToken, storeConfig);
+
+        const braintreeVisaCheckout = await this.braintreeSdk.getBraintreeVisaCheckout();
+
+        const visaCheckoutInstance = await this.braintreeSdk.getVisaCheckoutSdk(config.testMode);
+
+        const { container, onError } = visaCheckoutOptions;
+
+        const initOptions = braintreeVisaCheckout.createInitOptions({
+            paymentRequest: {
+                currencyCode: storeConfig.currency.code,
+                subtotal: String(checkout.subtotal),
+            },
+            settings: {
+                locale: storeConfig.storeProfile.storeLanguage,
+                shipping: {
+                    collectShipping: true,
+                },
+            },
+        });
+
+        const signInButton = this.createSignInButton(container, this.buttonClassName);
+
+        await visaCheckoutInstance.init(initOptions);
+
+        visaCheckoutInstance.on(
+            'payment.success',
+            (paymentSuccessPayload: VisaCheckoutPaymentSuccessPayload) =>
+                this.paymentSuccess(braintreeVisaCheckout, paymentSuccessPayload)
+                    .then(() => this.paymentIntegrationService.loadCheckout())
+                    .catch((error) => onError?.(error)),
+        );
+
+        visaCheckoutInstance.on('payment.error', (_, error) => onError?.(error));
+
+        signInButton.style.visibility = 'visible';
+    }
+
+    async deinitialize(): Promise<void> {
+        await this.braintreeSdk.deinitialize();
+    }
+
+    signIn(): Promise<void> {
+        throw new NotImplementedError(
+            'In order to sign in via VisaCheckout, the shopper must click on "Visa Checkout" button.',
+        );
+    }
+
+    async signOut(options?: RequestOptions): Promise<void> {
+        await this.paymentIntegrationService.signOutCustomer(options);
+
+        return Promise.resolve();
+    }
+
+    executePaymentMethodCheckout(options?: ExecutePaymentMethodCheckoutOptions): Promise<void> {
+        options?.continueWithCheckoutCallback?.();
+
+        return Promise.resolve();
+    }
+
+    private async paymentSuccess(
```

Comment:
It is hard to understand whats going on in this method looking for its naming. May I ask you to rename it with meaningful naming?

Reply:
Renamed